DSLink/DSLinkContainer.cs

```diff
@@ -1,61 +1,57 @@
 using System;
 using System.Threading.Tasks;
 using DSLink.Connection;
-using DSLink.Connection.Serializer;
+using DSLink.Serializer;
 using DSLink.Container;
 using DSLink.Util.Logger;
 using Newtonsoft.Json.Linq;
 using DSLink.Platform;
+using DSLink.Request;
+using DSLink.Respond;
 
 namespace DSLink
 {
-    /// <summary>
-    /// DSLink implementation of a container.
-    /// </summary>
-    // ReSharper disable once InconsistentNaming
     public class DSLinkContainer : AbstractContainer
     {
-        /// <summary>
-        /// Task used to send pings across the communication layer.
-        /// </summary>
         private readonly Task _pingTask;
-
-        /// <summary>
-        /// SerializationManager handles serialization for communications.
-        /// </summary>
-        internal SerializationManager SerializationManager;
-
-        /// <summary>
-        /// Handshake object, which contains data from the initial connection handshake.
-        /// </summary>
-        protected Handshake Handshake;
-
-        /// <summary>
-        /// Whether to reconnect to the broker. 
-        /// </summary>
-        internal bool Reconnect;
-
-        /// <summary>
-        /// Flag for when this DSLink Container is initialized.
-        /// Used to prevent duplicate initializations.
-        /// </summary>
-        private bool _isInitialized;
-
-        /// <summary>
-        /// DSLinkContainer constructor.
-        /// </summary>
-        /// <param name="config">Configuration for the DSLink</param>
-        public DSLinkContainer(Configuration config) : base(config)
+        protected Handshake ProtocolHandshake;
+        internal bool ReconnectOnFailure;
+        private bool _isLinkInitialized;
+        private readonly Configuration _config;
+        private readonly DSLinkResponder _responder;
+        private readonly DSLinkRequester _requester;
+        private readonly Connector _connector;
+        private readonly BaseLogger _logger;
+        private BaseSerializer _serializer;
+
+        public override Configuration Config => _config;
+        public override Responder Responder => _responder;
+        public override DSLinkRequester Requester => _requester;
+        public override Connector Connector => _connector;
+        public override BaseLogger Logger => _logger;
+        public override BaseSerializer DataSerializer => _serializer;
+
+        public DSLinkContainer(Configuration config)
         {
-            Logger = CreateLogger("DSLink");
+            _config = config;
+            _config._processOptions();
+            _logger = BasePlatform.Current.CreateLogger("DSLink", Config.LogLevel);
+            ReconnectOnFailure = true;
+            _connector = BasePlatform.Current.CreateConnector(this);
 
-            Reconnect = true;
-            Connector = BasePlatform.Current.CreateConnector(this);
+            if (Config.Responder)
+            {
+                _responder = new DSLinkResponder(this);
```

Comment:
Not sure I like the direction of that dependency. Does it make it hard to test to have the Responder/Requester requiring a DSLinkContainer?